File: arm.64/langs/zig/src/find.zig
const std = @import("std");

pub fn find(allocator: std.mem.Allocator, target_dir: ?[]const u8) !std.ArrayList([]const u8) {
    var files: std.ArrayList([]const u8) = .empty;

    var dir_path: []const u8 = "dist";
    if (target_dir) |td| {
        dir_path = td;
    } else if (std.process.getEnvVarOwned(allocator, "DIST_DIR")) |env_dir| {
        dir_path = env_dir;
    } else |_| {
        if (std.fs.cwd().openDir("dist", .{})) |_| {
            dir_path = "dist";
        } else |_| {
            if (std.fs.cwd().openDir("../../dist", .{})) |_| {
                dir_path = "../../dist";
            } else |_| {
                dir_path = "../../../dist";
            }
        }
    }

    var dir = std.fs.cwd().openDir(dir_path, .{ .iterate = true }) catch return files;
    defer dir.close();

    var iter = dir.iterate();
    while (try iter.next()) |entry| {
        if (entry.kind == .file and !std.mem.eql(u8, entry.name, ".DS_Store")) {
            const p = try std.fs.path.join(allocator, &[_][]const u8{ dir_path, entry.name });
            try files.append(allocator, p);
        }
    }
    return files;
}

File: arm.64/langs/cpp/src/find.cpp
#include "type.h"
#include <filesystem>
#include <cstdlib>

namespace fs = std::filesystem;

std::vector<std::string> find(const std::string& target_dir) {
    std::string dir = target_dir;
    if (dir.empty()) {
        if (const char* env_dir = std::getenv("DIST_DIR"); env_dir && *env_dir) {
            dir = env_dir;
        } else {
            std::vector<std::string> candidates = {
                (fs::current_path() / "dist").string(),
                (fs::current_path() / "../../dist").string(),
                (fs::current_path() / "../../../dist").string(),
                "dist"
            };
            for (const auto& c : candidates) {
                if (fs::exists(c) && fs::is_directory(c)) {
                    dir = c;
                    break;
                }
            }
        }
    }
    if (dir.empty() || !fs::exists(dir)) return {};

    std::vector<std::string> files;
    for (const auto& entry : fs::directory_iterator(dir)) {
        if (entry.is_regular_file() && entry.path().filename() != ".DS_Store") {
            files.push_back(entry.path().string());
        }
    }
    return files;
}
